test.sh

```diff
@@ -0,0 +1,455 @@
+}
+
+souffle_tests6() {
+    (cd "${THIS_DIR}/test" && ./run-souffle-tests-in-batches.py 125 149)
+}
+
+souffle_tests7() {
+    (cd "${THIS_DIR}/test" && ./run-souffle-tests-in-batches.py 150 175)
+}
+
+# 'd3log' test group.
+
+tcp_channel() {
+    #ZOOKEEPER_ENDPOINTS="127.0.0.1:2181"
+    #/usr/share/zookeeper/bin/zkServer.sh start &&
+    (cd "${THIS_DIR}/rust/template/distributed_datalog" && cargo fmt -- --check) &&
+    (cd "${THIS_DIR}/rust/template/distributed_datalog" && cargo clippy -- -D warnings) &&
+    #for i in $(seq 100); do
+    #    /usr/share/zookeeper/bin/zkServer.sh status && break;
+    #    sleep 1
+    #done &&
+    (cd "${THIS_DIR}/rust/template/distributed_datalog" && (
+        i=0;
+        true;
+        while [ $? -eq 0 -a $i -lt 100 ]; do
+          i=$((i+1));
+          cargo test -- tcp_channel::;
+        done
+        )
+    )
+}
+
+server_api() {
+    # It seems that stale files cause cargo to rebuild the project
+    # unnecessarily in CI.
+    #rm -rf test/datalog_tests/server_api_ddlog
+    (export DDLOG_HOME="${THIS_DIR}" && "${THIS_DIR}/test/datalog_tests/test-server_api.sh")
+}
+
+# 'stack' test group.
+
+modules() {
+    (cd "${THIS_DIR}" && stack --no-terminal test --ta "-p modules")
+}
+
+ovn_ftl() {
+    (cd "${THIS_DIR}" && stack --no-terminal test --ta "-p ovn_ftl")
+}
+
+ovn_mockup() {
+    (cd "${THIS_DIR}" && stack test --ta '-p "$(NF) == \"generate ovn\" || ($(NF-1) == \"compiler tests\" && $(NF) == \"ovn\")"')
+}
+
+redist() {
+    (cd "${THIS_DIR}" && STACK_CARGO_FLAGS='--release' stack test --ta '-p "$(NF) == \"generate redist\" || ($(NF-1) == \"compiler tests\" && $(NF) == \"redist\")"')
+}
+
+# 'misc' test group.
+
+span_string() {
+    (cd "${THIS_DIR}/test/datalog_tests" && ./run-test.sh span_string release)
+}
+
+span_uuid() {
+    (cd "${THIS_DIR}/test/datalog_tests" && ./run-test.sh span_uuid release)
+}
+
+path() {
+    (cd "${THIS_DIR}/test/datalog_tests" && ./run-test.sh path release)
+}
+
+#==========================================
+# Main test script.
+#==========================================
+
+printf "DDlog test suite\n\n"
+testsuite_setup
+
+if ( [ "$#" -eq 0 ] || [ "x$1" == "xhelp" ] || [ "x$1" == "x--help" ] ); then
+    printf "Usage: ${0} test_or_test_group1 test_or_test_group2 ...\n"
+    printf "   or: ${0} all\n\n"
+    printf "Available test groups:\n"
+    for group_with_descr in "${test_groups[@]}"
+    do
+        IFS=":" read -ra tokens <<< "$group_with_descr"
+        group=${tokens[0]}
+        description=${tokens[1]}
+
+        printf "\n    ${group}: ${description}\n"
+
+        eval "tests=(\"\${${group}[@]}\")"
+        for tst_with_descr in "${tests[@]}"
+        do
+            IFS=":" read -ra tokens <<< "$tst_with_descr"
+            tst=${tokens[0]}
+            description=${tokens[1]}
+            printf "        %-25s %s\n" "${tst}" "${description}"
+        done
+    done
+else
+    if [ "x$1" == "xall" ]; then
+        test_list=()
```

Comment:
maybe you should call this group_list

Reply:
It's really a mix of individual test names and group names.